source/common/http/conn_manager_utility.cc

```diff
@@ -101,11 +101,11 @@ void ConnectionManagerUtility::mutateRequestHeaders(
     request_headers.insertEnvoyDownstreamServiceCluster().value(config.userAgent().value());
     HeaderEntry& user_agent_header = request_headers.insertUserAgent();
     if (user_agent_header.value().empty()) {
-      user_agent_header.value(config.userAgent().value());
+      user_agent_header.value().setStatic(config.userAgent().value());
     }
 
     if (!local_info.nodeName().empty()) {
-      request_headers.insertEnvoyDownstreamServiceNode().value(local_info.nodeName());
+      request_headers.insertEnvoyDownstreamServiceNode().value().setStatic(local_info.nodeName());
```

Comment:
Yeah, I admit it's pretty hard to infer this beyond relying on ASAN (I think it would even be hard to ASSERT, since some of the "static" allocations will likely end up on the heap). If you think it's not worth it I can revert all of this.